README.md

```diff
@@ -0,0 +1,744 @@
+We're going to use a simple NodeJS application for testing. I've pushed image to DockerHub already (evalle/gordon:v1.0) a but you should never-ever download and run unknown images from DockerHub, so let's build them instead.
+
+Here is the app:
+```
+const http = require('http');
+const os = require('os');
+
+console.log("Gordon server starting...");
+
+var handler = function(request, response) {
+  console.log("Received request from " + request.connection.remoteAddress);
+  response.writeHead(200);
+  response.end("You've hit gordon v1, my name is " + os.hostname() + "\n");
+};
+
+var www = http.createServer(handler);
+www.listen(8080);
+```
+If you're not familiar with NodeJS, this app basically answers with greetings and hostname to any request to port 8080.
+
+Save it as `app-v1.js` and let's create a Dockerfile for it:
+```
+FROM node:10.5-slim
+ADD app-v1.js /app-v1.js
+ENTRYPOINT ["node", "app-v1.js"]
+```
+now build it:
+```
+$ docker build -t <your_name>/gordon:v1.0
+```
+And push it either to DockerHub or to your favorite Docker registry.
+
+### RC and Service
+Now let's create a ReplicationController and a service: they will be used by Ingress:
+```
+$ cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: gordon-v1
+spec:
+  replicas: 3
+  template:
+    metadata:
+      name: gordon-v1
+      labels:
+        app: gordon-v1
+    spec:
+      containers:
+      - image: evalle/gordon:v1.0
+        name: nodejs
+        imagePullPolicy: Always
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: gordon-service-v1
+spec:
+  selector:
+    app: gordon-v1
+  ports:
+  - port: 80
+    targetPort: 8080
+EOF
+```
+Don’t forget to change the image in the YAML file to your docker image:
+
+```
+...
+containers:
+- image: <name_of_your_image_goes_here>
+  name: nodejs
+...
+```
+Check that service is up and running:
+
+```
+$  kubectl get svc
+NAME                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
+gordon-service-v1   ClusterIP   10.111.33.157   <none>        80/TCP    1d
+kubernetes          ClusterIP   10.96.0.1       <none>        443/TCP   6d
+```
+
+Great, now let's create our first version of ingress:
+
+```
+$ cat <<EOF | kubectl create -f -
+kind: Ingress
+metadata:
+  name: gordon
+spec:
+  rules:
+  # this ingress maps the gordon.example.com domain name to our service
+  # you have to add gordon.example.com to /etc/hosts
+  - host: gordon.example.com
+    http:
+      paths:
+      # all requests will be sent to port 80 of the gordon service
+      - path: /v1
+        backend:
+          serviceName: gordon-service-v1
+          servicePort: 80
+EOF
+```
+To be able to access the service through the hostname gordon.example.com, we’ll need to make sure the domain name resolves to the IP of the ingress controller. We can do it via adding the hostname gordon.example.com into /etc/hosts or, in case of production applications we will need to set up DNS resolving properly.
```

Comment:
why don't use nip.io?

Reply:
I never used nip.io, and I don't know if every minikube will have the same ip address.

Reply:
/etc/hosts is ugly and we need to provide a cleanup afer the workshop.. nip.io is transparent and we can explain it.

Reply:
got ya, added nip.io option